components/script/dom/websocket.rs

```diff
@@ -498,10 +497,13 @@ impl Runnable for ConnectionEstablishedTask {
         };
 
         // Step 5: Cookies.
-        if let Some(cookies) = self.headers.get::<hyper::header::SetCookie>() {
-            let cookies = cookies.iter().map(|c| Serde(c.clone())).collect();
+        if let Some(cookies) =  self.headers.get::<hyper::header::SetCookie>() {
+            let mut v = String::new();
+            for cookie in cookies.iter() {
+                v.push_str(&cookie.to_string());
+            }
             let _ = ws.global().core_resource_thread().send(
-                SetCookiesForUrl(ws.url.clone(), cookies, HTTP));
+                SetCookiesForUrl(ws.url.clone(), v, HTTP));
```

Comment:
Could you use `SetCookiesForUrlWithData` instead, parsing each `String` from the `SetCookie` header to a `Cookie` before sending it? I would rather not roundtrip through a longer string only to split it again afterwards.